collector/cluster_health.go

```diff
@@ -220,49 +217,14 @@ func (c *ClusterHealth) Describe(ch chan<- *prometheus.Desc) {
 	ch <- c.jsonParseFailures.Desc()
 }
 
-func (c *ClusterHealth) fetchAndDecodeClusterHealth() (clusterHealthResponse, error) {
-	var chr clusterHealthResponse
-
-	u := *c.url
-	u.Path = "/_cluster/health"
-	res, err := c.client.Get(u.String())
-	if err != nil {
-		return chr, fmt.Errorf("failed to get cluster health from %s://%s:%s/%s: %s",
-			u.Scheme, u.Hostname(), u.Port(), u.Path, err)
-	}
-	defer res.Body.Close()
-
-	if res.StatusCode != http.StatusOK {
-		return chr, fmt.Errorf("HTTP Request failed with code %d", res.StatusCode)
-	}
-
-	if err := json.NewDecoder(res.Body).Decode(&chr); err != nil {
-		c.jsonParseFailures.Inc()
-		return chr, err
-	}
-
-	return chr, nil
-}
-
-func (c *ClusterHealth) Collect(ch chan<- prometheus.Metric) {
+func (c *ClusterHealth) Collect(ch chan<- prometheus.Metric, clusterHealthResponse clusterHealthResponse) {
```

Comment:
this doesn't work. Collect must fulfill the prometheus. Collector interface and thus the signature must be
```
Collect(ch chan<- prometheus.Metric)
```